### ISSUE TITLE
@libsql/client embedded replica leaks memory

### ISSUE THREAD

**Issue by khuezy** [libsql/libsql-client-ts#135]
Hi! There is a major memory leak when using embedded replicas (calling client.sync and then making any query on the client).
The reproduction repo is [here](https://github.com/khuezy/turso_leak) with instructions.

Here is a screenshot of the Fly container memory (using sync)
![image](https://github.com/libsql/libsql-client-ts/assets/811298/99f0f647-17a8-4801-97a8-aeeba4120318)

Here is a screenshot of the same app but w/o using embedded replicas:
![image (1)](https://github.com/libsql/libsql-client-ts/assets/811298/de9a08db-97a6-4137-9546-499b4486fe11)

**Comment by LucioFranco** [comment]
cc @psarna @MarinPostma

**Comment by MarinPostma** [comment]
maybe linked to: https://ptb.discord.com/channels/933071162680958986/1175813081717280900/1175813081717280900

**Comment by khuezy** [comment]
Yes that looks like it's related. Perhaps it's not embedded replicas, but a local file that causing the leak.
